### ISSUE TITLE
Network dataset update problem

### ISSUE THREAD

**Issue by kandangit** [almende/vis#3282]
Hi,
Have a network graph using a dataset.
This dataset contains nodes with values for their coordinates X, Y

Programmatically, I change a node´s colour or label

```js
       var node = nodesds.getDataSet().get(id);
       node["label"] = label;
       nodesds.getDataSet().update(node);
```

This however resets the node X, Y
workaround is to set X manually: `node["x"] = network.getPositions()[2].x;`

Is there a better solution? I thought datasets linked to the graph keep a refreshed view of the nodes-edges properties.

**Comment by wimrijnders** [comment]
Duplicate of #3096.

If you have x/y coordinates in your node-`DataSets`, they will be reset in the `Network` instance on every data update. This is a consequence of the design decision to leave the data in `DataSet` untouched during display. Which is a good decision by itself, but it leads to situations like these.

Your solution is OK; a perhaps better alternative is to strip out the xy-coordinates from the nodes-`Dataset` before update. This needs to be done only once, and after that you can update to your heart's content. Check out [this comment](https://github.com/almende/vis/issues/3096#issuecomment-305518575) on how to do this.

> I thought datasets linked to the graph keep a refreshed view of the nodes-edges properties.

Well, no, as indicated above. Data in the DataSet is user-controlled and will not change due to states in the `Network` instance. If I think about it, this is a good principle; I hope you can see this too.

**Comment by kandangit** [comment]
It makes sense to separate the model of the dataset and the view x,y coordinates.

In my canse have to get the node positions first then before updating any node. 

I store the graph configuration (x,y) and retreive it to show the graph in the saved configuration. For this I can remove the x,y properties. 

If a user updates a node label from a dialog, i do an update on the nodes dataset. Will also pass the x,y information.


Shall I close?

thanks

**Comment by wimrijnders** [comment]
You solution is perfectly sensible. Yes, please close if you are done with this issue. Cheers!
